### PR TITLE
windows dokanclean: Use registry breadcrumb

### PR DESCRIPTION
**Pull request by zanderz**
This is the change that will prevent us from uninstalling someone else's Dokan.
Also added a logger object.

@maxtaco @oconnor663

### CODE REVIEW

File: go/tools/dokanclean/main.go [hunk 1]
```diff
@@ -90,53 +89,66 @@ func findDokanUninstall(list bool, wow64 bool) (result int) {
...
-			if err != nil {
-				uninstall, _, err = subKey.GetStringValue("UninstallString")
-			}
-			if err != nil {
-				fmt.Printf("Error %s opening subkey UninstallString", err.Error())
-			} else {
-				result = mergeResults(result, doUninstallAction(uninstall, list))
-			}
+		uninstall, _, err := subKey.GetStringValue("QuietUninstallString")
+		if err != nil {
```
Comment (maxtaco) [inline go/tools/dokanclean/main.go]: Bug? 'Err == nil'?

**Comment by maxtaco** [comment]
The c/i failures on OSX are fixed in master, I think a rebase will fix them.

File: go/tools/dokanclean/main.go [hunk 2]
```diff
@@ -90,53 +89,66 @@ func findDokanUninstall(list bool, wow64 bool) (result int) {
...
+			uninstall, _, err = subKey.GetStringValue("UninstallString")
+		}
+		if err != nil {
+			log.Printf("Error opening subkey UninstallString: %s", err.Error())
+		} else {
+			result = mergeResults(result, doUninstallAction(uninstall, list, log))
```
Comment (maxtaco) [inline go/tools/dokanclean/main.go]: Uninstall should also remove the reigstry key that Dokan is installed.

**Comment by maxtaco** [comment]
Is there any good way to write some tests for this/

**Comment by maxtaco** [comment]
Made a pass through

File: go/tools/dokanclean/main.go (hunk 1 above)
Comment (zanderz) [inline go/tools/dokanclean/main.go]: The intent is for `UninstallString` to be the default, but use `QuietUninstallString` if available.

File: go/tools/dokanclean/main.go (hunk 2 above)
Comment (zanderz) [inline go/tools/dokanclean/main.go]: That's how [the installer treats this key](https://github.com/keybase/client/blob/master/packaging/windows/WIXInstallers/KeybaseApps/KeybaseApps.wxs#L104), but I will double check

**Comment by zanderz** [comment]
RE: testing - that is what the -l switch is for, more or less. I don't think it's a good idea to write to that part of the registry as a routine test.

File: go/tools/dokanclean/main.go (hunk 1 above)
Comment (maxtaco) [inline go/tools/dokanclean/main.go]: Ah ok

**Pull request merged**